crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -752,22 +754,28 @@ all_tuples_enumerated!(impl_param_set, 1, 8, P, p);
 // SAFETY: Res only reads a single World resource
 unsafe impl<'a, T: Resource> ReadOnlySystemParam for Res<'a, T> {}
 
-// SAFETY: Res ComponentId access is applied to SystemMeta. If this Res
-// conflicts with any prior access, a panic will occur.
+// SAFETY: ResMut component access is tied to the singleton components under
+// QueryState<Ref<T>, With<Internal>>, which guarantees safety.
```

Comment:
> which guarantees safety

Safety comments should explain which safety requirements are satisfied and how, this does neither.